src/main/java/ch/njol/skript/expressions/ExprName.java

```diff
@@ -189,6 +195,11 @@ public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
 		for (Object o : getExpr().getArray(e)) {
 			if (o instanceof Player) {
 				switch (mark) {
+					case 1:
+						Player p = (Player) o;
+						PlayerProfile profile = p.getPlayerProfile();
+						profile.setName(name);
```

Comment:
From what I can tell, this won't work for the RESET changer. Is it possible to get the real name of a player after it has been changed, or no?

If not, then you will need to disallow the reset changer for this